src/super_context.ml

```diff
@@ -270,29 +272,23 @@ module Libs = struct
                  raise (Lib.Error (No_solution_found_for_select e))
              }))
 
-  let requires t ~dir ~has_dot_merlin compile_info =
-    add_select_rules t ~dir (Lib.Compile.resolved_selects compile_info);
-    let requires =
-      Build.of_result_map (Lib.Compile.requires compile_info)
-        ~f:Build.return
-    in
-    let requires =
+  let with_lib_deps t compile_info ~dir ~has_dot_merlin ~f =
```

Comment:
I think you might be forgetting to set `~has_dot_merlin` correctly. I see it's being set to `true` everywhere rather than being based on the context.